persistence/trees/trees_test.go

```diff
@@ -1,111 +1,205 @@
+		Output:          hex.EncodeToString(outputAddr),
 	}
-	treeStore.merkleTrees["test"] = testTree
-
-	testCases := []struct {
-		name        string
-		treeName    string
-		key         []byte
-		value       []byte
-		valid       bool
-		expectedErr error
-	}{
-		{
-			name:        "valid inclusion proof: key and value in tree",
-			treeName:    "test",
-			key:         []byte("key"),
-			value:       []byte("value"),
-			valid:       true,
-			expectedErr: nil,
-		},
-		{
-			name:        "valid exclusion proof: key not in tree",
-			treeName:    "test",
-			key:         []byte("key2"),
-			value:       nil,
-			valid:       true,
-			expectedErr: nil,
-		},
-		{
-			name:        "invalid proof: tree not in store",
-			treeName:    "unstored tree",
-			key:         []byte("key"),
-			value:       []byte("value"),
-			valid:       false,
-			expectedErr: fmt.Errorf("tree not found: %s", "unstored tree"),
-		},
-		{
-			name:        "invalid inclusion proof: key in tree, wrong value",
-			treeName:    "test",
-			key:         []byte("key"),
-			value:       []byte("wrong value"),
-			valid:       false,
-			expectedErr: nil,
-		},
-		{
-			name:        "invalid exclusion proof: key in tree",
-			treeName:    "test",
-			key:         []byte("key"),
-			value:       nil,
-			valid:       false,
-			expectedErr: nil,
-		},
+}
+
+// TECHDEBT(#796): Test helpers should be consolidated in a single place
+func newTestPostgresContext(t testing.TB, height int64, testPersistenceMod modules.PersistenceModule) *persistence.PostgresContext {
+	t.Helper()
+	rwCtx, err := testPersistenceMod.NewRWContext(height)
+	if err != nil {
+		log.Fatalf("Error creating new context: %v\n", err)
 	}
 
-	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
-			valid, err := treeStore.Prove(tc.treeName, tc.key, tc.value)
-			require.Equal(t, valid, tc.valid)
-			if tc.expectedErr == nil {
-				require.NoError(t, err)
-				return
-			}
-			require.ErrorAs(t, err, &tc.expectedErr)
-		})
+	postgresCtx, ok := rwCtx.(*persistence.PostgresContext)
+	if !ok {
+		log.Fatalf("Error casting RW context to Postgres context")
 	}
+
+	// TECHDEBT: This should not be part of `NewTestPostgresContext`. It causes unnecessary resets
+	// if we call `NewTestPostgresContext` more than once in a single test.
+	t.Cleanup(func() {
+		resetStateToGenesis(testPersistenceMod)
+	})
+
+	return postgresCtx
+}
+
+// This is necessary for unit tests that are dependant on a baseline genesis state
+func resetStateToGenesis(m modules.PersistenceModule) {
+	if err := m.ReleaseWriteContext(); err != nil {
+		log.Fatalf("Error releasing write context: %v\n", err)
+	}
+	if err := m.HandleDebugMessage(&messaging.DebugMessage{
+		Action:  messaging.DebugMessageAction_DEBUG_PERSISTENCE_RESET_TO_GENESIS,
+		Message: nil,
+	}); err != nil {
+		log.Fatalf("Error clearing state: %v\n", err)
+	}
+}
+
+// TODO_AFTER(#861): Implement this test with the test suite available in #861
```

Comment:
`test suite available in #861` is this correct?

We want to use the test suite in this PR to implement it?

Reply:
I believe that's what @h5law meant out of this, yes.